internal/pkg/fieldtags/analyzer.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package fieldtags defines an analyzer that identifies struct fields tagged with datapol annotations
+// indicating that they contain SPII.
+package fieldtags
+
+import (
+	"go/ast"
+	"reflect"
+	"strings"
+
+	"golang.org/x/tools/go/analysis"
+	"golang.org/x/tools/go/analysis/passes/inspect"
+	"golang.org/x/tools/go/analysis/passes/structtag"
+	"golang.org/x/tools/go/ast/inspector"
+)
+
+type keyValue struct {
+	key   string
+	value string
+}
+
+type sourcePatterns []keyValue
+
+var patterns sourcePatterns = []keyValue{
+	{"levee", "source"},
+}
+
+var Analyzer = &analysis.Analyzer{
+	Name:       "fieldtags",
+	Doc:        "This analyzer identifies Source fields based on their tags.",
+	Run:        run,
+	Requires:   []*analysis.Analyzer{inspect.Analyzer, structtag.Analyzer},
+	ResultType: reflect.TypeOf(new([]*ast.Field)).Elem(),
+}
+
+func run(pass *analysis.Pass) (interface{}, error) {
+	inspect := pass.ResultOf[inspect.Analyzer].(*inspector.Inspector)
```

Comment:
Yikes. Good catch! I think originally I wanted to call it `inspector`, but I noticed the clash with the `inspector` package, so I renamed it to `inspect`... :upside_down_face:

Reply:
I renamed it to `inspector_`. I can't seem to think of a better name. Is using an underscore to avoid a name collision idiomatic in Go?

Reply:
Go eschews underscore naming.  While I personally dislike it, I think it would be idiomatic to name it with something abbreviated, such as `insp`.  If you want something more descriptive, `inspResult` or `inspectResult` could be fine.

Reply:
SG. Not a fan of abbreviations either. I'll go with `inspectResult`.